### PR TITLE
feat: linux add postinstall script

### PR DESCRIPTION
**Pull request by hiento09**
## Describe Your Changes

- feat: linux add postinstall script

## Fixes Issues

- Related ticket #1654 

image
image

### CODE REVIEW

File: electron/scripts/post-uninstall.sh [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+
+echo "The application is being removed. Executing afterRemove script."
+
+# Determine the home directory based on the user
+USER_TO_RUN_AS=${SUDO_USER:-$(whoami)}
+if [ "$USER_TO_RUN_AS" = "root" ]; then
+    USER_HOME="/root"
+else
+    USER_HOME="/home/$USER_TO_RUN_AS"
+fi
```
Comment (louis-jan) [inline electron/scripts/post-uninstall.sh:25]: Users can relocate Jan data folder

File: electron/scripts/post-uninstall.sh [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+            [yY][eE][sS]|[yY])
+                echo "Deleting jan data folders..."
+                if [ -d "$USER_HOME/$DATA_FOLDER_NAME" ]; then
+                    echo "Removing $USER_HOME/$DATA_FOLDER_NAME"
+                    rm -rf "$USER_HOME/$DATA_FOLDER_NAME" > /dev/null 2>&1
+                fi
+                break
+                ;;
+            [nN][oO]|[nN]|"")
+                echo "Keeping the 'jan' data folders."
```
Comment (louis-jan) [inline electron/scripts/post-uninstall.sh:43]: Copy @imtuyethan

File: electron/scripts/post-uninstall.sh (hunk 1 above)
Comment (louis-jan) [reply electron/scripts/post-uninstall.sh:25]: So it does not work for these 2 cases:  
- New users
- Users relocated Jan data folder

**Comment by hiento09** [comment]
- It does not work for these 2 cases:
- New users
- Users relocated Jan data folder

**Pull request closed without merging**